### HW Questions.

In [6]:
import duckdb
conn = duckdb.connect("ny_taxi.duckdb", read_only=True)

---

#### **Q1.** What is the start date and end date of the dataset?

In [7]:
conn.execute("SELECT MIN(trip_pickup_date_time), MAX(trip_pickup_date_time) FROM ny_taxi_data.rides").fetchall()

[(datetime.datetime(2009, 6, 1, 7, 33, tzinfo=<DstTzInfo 'America/New_York' EDT-1 day, 20:00:00 DST>),
  datetime.datetime(2009, 6, 30, 19, 58, tzinfo=<DstTzInfo 'America/New_York' EDT-1 day, 20:00:00 DST>))]

Answer: `2009-06-01 to 2009-07-01`

---

#### **Q2.** What proportion of trips are paid with credit card?

In [8]:
conn.execute("""
SELECT COUNT(*) FILTER (WHERE payment_type = 'Credit') * 100.0 / COUNT(*) 
FROM ny_taxi_data.rides
""").fetchall()

[(26.66,)]

Answer: `26.66%`

---

#### **Q3.** What is the total amount of money generated in tips?

In [9]:
conn.execute("SELECT SUM(tip_amt) FROM ny_taxi_data.rides").fetchall()

[(6063.410000000009,)]

Answer: `$6,063.41`

---